src/ws-dropdown-menu.js

```diff
@@ -0,0 +1,289 @@
+      items: props.items,
+      value: props.value,
+      openSubMenu: null
+    };
+  }
+
+  /**
+   * Gets the current height of the menu
+   * @returns {Number}
+   */
+  getHeight() {
+    return this.menuContainer.clientHeight;
+  }
+
+  /**
+   * Send the new height of this menu after update to the parent.
+   * This will be called when updateFilter did set the new state
+   * @void
+   */
+  componentDidUpdate() {
+    this.props.handle('change-size', this.getHeight());
+  }
+
+  /**
+   * Setting the input value as filter
+   * @param event {KeyboardEvent}
+   * @void
+   */
+  updateFilter(event) {
+    event.stopPropagation();
+    this.setState({filter: event.target.value});
+  }
+
+  /**
+   * If there is a filter active it applies it on the available items
+   * @returns {Array<Object>}
+   */
+  getFilteredItems() {
+    let regex = new RegExp(this.state.filter, 'i');
+    return this.state.items.filter(item => {
+      // Don't show items which doesn't match the filter
+      if (this.props.filterable && this.state.filter && !regex.test(item.label)) {
+        return false;
+      }
+      // Don't show items which were stored as value -> don't show them twice
+      return this.context.multiple ? !item.stored : true;
+    });
+  }
+
+  /**
+   * Deselect all items which are not stored as value. Only relevant for multi select dropdown.
+   * When the dropdown will be closed the state will be restored
+   * @void
+   */
+  clearSelections() {
+    if (this.state.items) {
+      this.state.items.forEach(item => {
+        if (item.selected && !item.stored) {
+          item.selected = false;
+        }
+      });
+    }
+  }
+
+  /**
+   * Handles submit action on multi select drop downs
+   * @void
+   */
+  submit(event) {
+    event.stopPropagation();
+    this.state.value = this.state.items.filter(item => {
+      item.stored = item.selected;
+      return item.selected;
+    });
+    // Propagate new value
+    this.props.handle('change', this.state.value);
+    this.setState(this.state);
+  }
+
+  /**
+   * Handles data propagation from child menus
+   * @param type {String} Should be just show-parent
+   * @param data {*}
+   */
+  handlePropagation = (type, data) => {
+    switch (type) {
+      // This comes from an dropdown-item and has to be propagated to the parent
+      case 'go-back':
+        this.props.handle('show-parent');
+        break;
+      // This comes from the child menu
+      case 'show-parent':
+        this.showCurrent();
+        break;
+      case 'show-child':
+        this.showChild(data);
+        break;
+      case 'change':
+        this.clearSelections();
+        this.props.handle(type, data);
+        break;
+      case 'change-size':
+        this.props.handle(type, data);
+        break;
+    }
+  };
+
+  /**
+   * Shows the child menu and hides the current menu
+   * @param subMenu {WSDropdownMenu}
+   * @void
+   */
+  showChild(subMenu) {
+    this.state.openSubMenu = subMenu;
```

Comment:
Here I think it'd work if we use nested refs ex. `this.refs.WSDropdownItem.refs.openMenu` I'm not sure if we'd need state for this.
If there's no way around it, I'd still suggest using this.setState() and not mutate the value directly